Assets/Scripts/Controllers/BuildModeController.cs

```diff
@@ -251,43 +251,43 @@ public bool DoesBuildJobOverlapExistingBuildJob(Tile t, string furnitureType)
         return false;
     }
 
-    public void DeconstructFurnitureAt(Tile t, string nextBuild = null)
+    public void DeconstructFurnitureAt(Tile tile, string nextBuild = null)
     {
-        Job j;
+        Job job;
 
-        if (PrototypeManager.FurnitureJobDeconstruct.HasPrototype(t.Furniture.ObjectType))
+        if (PrototypeManager.FurnitureJobDeconstruct.HasPrototype(tile.Furniture.ObjectType))
         {
             // Make a clone of the job prototype
-            j = PrototypeManager.FurnitureJobDeconstruct.GetPrototype(t.Furniture.ObjectType).Clone();
+            job = PrototypeManager.FurnitureJobDeconstruct.GetPrototype(tile.Furniture.ObjectType).Clone();
 
             // Assign the correct tile.
-            j.tile = t;
-            j.OnJobCompleted += (Job jo) =>
+            job.tile = tile;
+            job.OnJobCompleted += (Job jo) =>
```

Comment:
This would lead to `job` not collected by GC you need to detach from event after you finished.